chrome/content/zotero/elements/attachmentPreview.js

```diff
@@ -177,13 +178,47 @@
 			this.removeEventListener("click", this._handleFocusIn);
 			this.removeEventListener("focusin", this._handleFocusIn);
 			this.removeEventListener("keypress", this._handleKeypress);
+			Zotero.Notifier.unregisterObserver(this._notifierID);
+		}
+
+		notify(event, type, ids, extraData) {
+			if (!this.item) return;
+			if (this.isReaderType && this._reader) {
+				// Following chrome/content/zotero/xpcom/reader.js
+				if (event === "delete") {
+					let disappearedIDs = this._reader.annotationItemIDs.filter(x => ids.includes(x));
+					if (disappearedIDs.length) {
+						let keys = disappearedIDs.map(id => extraData[id].key);
+						this._reader.unsetAnnotations(keys);
+					}
+				}
+				else if (["add", "modify"].includes(event)) {
+					let annotationItems = this.item.getAnnotations();
+					this._reader.annotationItemIDs = annotationItems.map(x => x.id);
+					let affectedAnnotations = annotationItems.filter(({ id }) => (
+						ids.includes(id)
+						&& !(extraData && extraData[id] && extraData[id].instanceID === this._reader._instanceID)
+					));
+					if (affectedAnnotations.length) {
+						this._reader.setAnnotations(affectedAnnotations);
+					}
+				}
+				return;
+			}
+			if (this.isMediaType) {
+				if (["refresh", "modify"].includes(event) && ids.includes(this.item.id)) {
+					this.discard().then(() => this.render());
+				}
+			}
 		}
 
 		async render() {
 			let itemID = this._item?.id;
 			if (!this.initialized && itemID === this._renderingItemID) {
 				return;
 			}
+			// For tests
+			this._renderDeferred = Zotero.Promise.defer();
```

Comment:
If this is only for tests (and there's no way to avoid), we can put in `if (Zotero.test) {` to make sure that nothing is relying on it that shouldn't be.

Reply:
This is also technically a deprecated pattern from Bluebird that we don't really want to use. Can use native promises instead:

```javascript
let resolve;
this._renderPromise = new Promise((res) => {
    resolve = res;
}));
```

And below just:

```javascript
resolve();
```

And in tests:

```javascript
await itemDetails._renderPromise;
```

Reply:
done